relax version pin on mio ceate in supervisor

As of the date this issue was opened mio is currently at 0.7.0. This includes some breaking changes. Most are fairly straight forward however other crates are brining in 0.6.21 as well and I have not been able to get around:

```
error[E0599]: no function or associated item named `new` found for type `mio::poll::Poll` in the current scope
   --> components\sup\src\manager\service\pipe_hook_client.rs:134:26
    |
134 |         let poll = Poll::new()?;
    |                          ^^^ function or associated item not found in `mio::poll::Poll`
```

--- a/components/sup/Cargo.toml
+++ b/components/sup/Cargo.toml
@@ -75,12 +75,9 @@ jemalloc-ctl = "*"
 [target.'cfg(windows)'.dependencies]
 ctrlc = "*"
 habitat-launcher-protocol = { path = "../launcher-protocol" }
-mio-named-pipes = "*"
-# Pinning for now. Since upgrading to 0.7.0 results in conflicts with other crates
-# See https://github.com/habitat-sh/habitat/issues/7522
-mio = "^0.6"
+mio = { version = "*", features = ["os-ext"] }
 uuid = { version = "*", features = ["v4"] }
-winapi =  { version = "^0.3.9", features = ["namedpipeapi", "tlhelp32"] }
+winapi = { version = "^0.3.9", features = ["namedpipeapi", "tlhelp32"] }
 
 [dev-dependencies]
 habitat_core = { path = "../core" }

--- a/components/sup/src/manager/service/pipe_hook_client.rs
+++ b/components/sup/src/manager/service/pipe_hook_client.rs
@@ -7,12 +7,11 @@ use habitat_core::{env as henv,
                    os::process::windows_child::Child,
                    util::{self,
                           BufReadLossy}};
-use mio::{Events,
+use mio::{windows::NamedPipe,
+          Events,
+          Interest,
           Poll,
-          PollOpt,
-          Ready,
           Token};
-use mio_named_pipes::NamedPipe;
 use std::{self,
           env,
           ffi::OsStr,
@@ -32,7 +31,8 @@ use std::{self,
           time::{Duration,
                  Instant}};
 use uuid::Uuid;
-use winapi::um::{namedpipeapi,
+use winapi::um::{handleapi,
+                 namedpipeapi,
                  processthreadsapi,
                  winbase};
 
@@ -89,30 +89,40 @@ impl PipeHookClient {
         File::create(&self.stdout_log_file)?;
         File::create(&self.stderr_log_file)?;
 
-        let (mut pipe, poll) = self.connect()?;
+        let (mut pipe, mut poll) = self.connect()?;
         debug!("connected to {} {} hook pipe",
                service_group, self.hook_name);
 
         // The powershell server takes a single byte as input which will be either
         // 0 to shut down (see drop below) or 1 to run the hook
-        self.pipe_ready(&poll, Ready::writable())?;
+        self.pipe_ready(&mut poll, Interest::WRITABLE)?;
         pipe.write_all(&SIGNAL_EXEC_HOOK)?;
 
         // Now we wait for the hook to run and the powershell service to
         // send back the hook's exit code over the pipe
-        self.pipe_ready(&poll, Ready::readable())?;
+        self.pipe_ready(&mut poll, Interest::READABLE)?;
         let mut exit_buf = [0; std::mem::size_of::<u32>()];
         pipe.read_exact(&mut exit_buf)?;
+        unsafe {
+            handleapi::CloseHandle(pipe.as_raw_handle());
+        }
         Ok(u32::from_ne_bytes(exit_buf))
     }
 
-    fn pipe_ready(&self, poll: &Poll, readiness: Ready) -> io::Result<bool> {
+    fn pipe_ready(&self, poll: &mut Poll, readiness: Interest) -> io::Result<bool> {
         let mut events = Events::with_capacity(1024);
         let loop_value = loop {
             let checked_thread = liveliness_checker::mark_thread_alive();
-            let result =
-                poll.poll(&mut events, None)
-                    .map(|_| events.iter().any(|e| e.readiness().contains(readiness)));
+            let result = poll.poll(&mut events, None).map(|_| {
+                                                         events.iter().any(|e| {
+                                                                          (e.is_readable()
+                                                                  && readiness
+                                                                     == Interest::READABLE)
+                                                                 || (e.is_writable()
+                                                                     && readiness
+                                                                        == Interest::WRITABLE)
+                                                                      })
+                                                     });
             if let Ok(false) = result {
                 continue;
             } else {
@@ -131,9 +141,10 @@ impl PipeHookClient {
             .custom_flags(winbase::FILE_FLAG_OVERLAPPED);
         let file = opts.open(self.abs_pipe_name())?;
 
-        let pipe = unsafe { NamedPipe::from_raw_handle(file.into_raw_handle()) };
+        let mut pipe = unsafe { NamedPipe::from_raw_handle(file.into_raw_handle()) };
         let poll = Poll::new()?;
-        poll.register(&pipe, Token(0), Ready::all(), PollOpt::edge())?;
+        poll.registry()
+            .register(&mut pipe, Token(0), Interest::WRITABLE | Interest::READABLE)?;
         Ok((pipe, poll))
     }
 
@@ -231,8 +242,8 @@ impl PipeHookClient {
             debug!("error checking if pipe exists: {}", err);
         } else {
             debug!("Telling {} pipe server to quit", self.pipe_name);
-            let (mut pipe, poll) = self.connect()?;
-            self.pipe_ready(&poll, Ready::writable())?;
+            let (mut pipe, mut poll) = self.connect()?;
+            self.pipe_ready(&mut poll, Interest::WRITABLE)?;
             pipe.write_all(&SIGNAL_QUIT)?;
         }
         Ok(())

